grid.v3/mailbox.go

```diff
@@ -25,31 +22,55 @@ func (box *Mailbox) String() string {
 
 // NewMailbox for requests addressed to name. Size will be the mailbox's
 // channel size.
-func NewMailbox(c context.Context, name string, size int) (*Mailbox, error) {
+//
+// Example Usage:
+//
+//     func Act(c context.Context) {
+//         mailbox, err := NewMailbox(c, "incoming", 10)
+//         ...
+//         defer mailbox.Close()
+//
+//         for {
+//             select {
+//             case req := <-mailbox.C:
+//                 // Do something with request, and then respond
+//                 // or ack. A response or ack is required.
+//                 switch m := req.Msg().(type) {
+//                 case *HiMsg:
+//                     req.Respond(&HelloMsg{...})
+//                 }
+//             }
+//         }
+//     }
+//
+// If the mailbox has already been created, in the calling process or
+// any other process, an error is returned, since only one mailbox
+// can claim a particular name.
+//
+// Using a mailbox requires that the process creating the mailbox also
+// started a grid Server.
+func NewMailbox(s *Server, name string, size int) (*Mailbox, error) {
+	if !isServerRunning(s) {
+		return nil, ErrServerNotRunning
+	}
 	if !isNameValid(name) {
 		return nil, ErrInvalidMailboxName
 	}
 
```

Comment:
reminds me of matchbox: https://dzone.com/articles/lose-the-lock-while-embracing-concurrency-brave-ne